### PR TITLE
Add more ChainRules derivatives

### PR DESCRIPTION
**Pull request by devmotion**
This PR adds and tests derivatives for
- `airyaix`
- `airyaiprimex`
- `logerfc`
- `logerfcx`
- `erf(x, y)`
- `besselkx`
- `hankelh1x`
- `hankelh2x`

It also replaces irrationals in derivatives with constants from IrrationalConstants. This fixes https://github.com/JuliaMath/SpecialFunctions.jl/issues/307.

### CODE REVIEW

**Review by stevengj — APPROVED** [review]

File: src/chainrules.jl [hunk 1]
```diff
@@ -31,12 +33,15 @@ ChainRulesCore.@scalar_rule(
 )
 ChainRulesCore.@scalar_rule(dawson(x), 1 - (2 * x * Ω))
 ChainRulesCore.@scalar_rule(digamma(x), trigamma(x))
-ChainRulesCore.@scalar_rule(erf(x), (2 / sqrt(π)) * exp(-x * x))
-ChainRulesCore.@scalar_rule(erfc(x), -(2 / sqrt(π)) * exp(-x * x))
+ChainRulesCore.@scalar_rule(erf(x), (2 / sqrt(π)) * exp(-x^2))
+ChainRulesCore.@scalar_rule(erfc(x), -(2 / sqrt(π)) * exp(-x^2))
+ChainRulesCore.@scalar_rule(logerfc(x), -(2 / sqrt(π)) * exp(-x^2 - Ω))
```
Comment (stevengj) [inline src/chainrules.jl:38]: Would be good to use the new constants from #345.

Comment (devmotion) [reply src/chainrules.jl:38]: I'll update the PR now that the other one was merged.

Comment (devmotion) [reply src/chainrules.jl]: I updated the ChainRules definitions, they use the irrational constants as well now.

This fixes https://github.com/JuliaMath/SpecialFunctions.jl/issues/307 for most functions. However, for `erfcx` additional constants would be needed: https://github.com/JuliaMath/IrrationalConstants.jl/pull/8

File: src/chainrules.jl [hunk 2]
```diff
@@ -31,12 +33,18 @@ ChainRulesCore.@scalar_rule(
...
-ChainRulesCore.@scalar_rule(erfi(x), (2 / sqrt(π)) * exp(x * x))
-ChainRulesCore.@scalar_rule(erfinv(x), (sqrt(π) / 2) * exp(Ω^2))
+
+# TODO: use `invsqrtπ` if it is added to IrrationalConstants
+ChainRulesCore.@scalar_rule(erf(x), (2 * exp(-x^2)) / sqrtπ)
+ChainRulesCore.@scalar_rule(erf(x, y), (- (2 * exp(-x^2)) / sqrtπ, (2 * exp(-y^2)) / sqrtπ))
+ChainRulesCore.@scalar_rule(erfc(x), - (2 * exp(-x^2)) / sqrtπ)
+ChainRulesCore.@scalar_rule(logerfc(x), - (2 * exp(-x^2 - Ω)) / sqrtπ)
+ChainRulesCore.@scalar_rule(erfcinv(x), - (sqrtπ * (exp(Ω^2) / 2)))
+ChainRulesCore.@scalar_rule(erfcx(x), 2 * x * Ω - 2 / sqrtπ)
```
Comment (stevengj) [inline src/chainrules.jl:43]:
```suggestion
ChainRulesCore.@scalar_rule(erfcx(x), 2 * x * Ω - 2 / oftype(Ω, sqrtπ))
```

Comment (devmotion) [reply src/chainrules.jl]: That's a good workaround until `invsqrtπ` is available :+1: I also updated the tests.

**Comment by devmotion** [comment]
Maybe it's time for a new release (possibly after merging the fixes in https://github.com/JuliaMath/SpecialFunctions.jl/pull/347)? The version number is already updated on the master branch.

**Pull request merged**